pkg/tfbridge/schema.go

```diff
@@ -315,6 +316,37 @@ func MakeTerraformInputs(
 		ComputeDefaultOptions: cdOptions,
 		ProviderConfig:        config,
 		ApplyDefaults:         true,
+		ApplyTFDefaults:       true,
+		Assets:                AssetTable{},
+	}
+	inputs, err := cctx.makeTerraformInputs(olds, news, tfs, ps)
+	if err != nil {
+		return nil, nil, err
+	}
+	return inputs, cctx.Assets, err
+}
+
+func MakeTerraformInputsNoTFDefaults(
```

Comment:
`MakeTerraformInputsNoTFDefaults` is identical to `MakeTerraformInputs` except with `ApplyTFDefaults: true` instead of `false`. Let's keep it DRY by factoring out the helper method:

```go
func makeTerraformInputsHelper(
	ctx context.Context, instance *PulumiResource, config resource.PropertyMap,
	olds, news resource.PropertyMap, tfs shim.SchemaMap, ps map[string]*SchemaInfo,
        applyTFDefaults bool,
) (map[string]interface{}, AssetTable, error) {
	cdOptions := ComputeDefaultOptions{}
	if instance != nil {
		cdOptions = ComputeDefaultOptions{
			PriorState: olds,
			Properties: instance.Properties,
			Seed:       instance.Seed,
			URN:        instance.URN,
		}
	}

	cctx := &conversionContext{
		Ctx:                   ctx,
		ComputeDefaultOptions: cdOptions,
		ProviderConfig:        config,
		ApplyDefaults:         true,
		ApplyTFDefaults:       true,
		Assets:                AssetTable{},
	}

	inputs, err := cctx.makeTerraformInputs(olds, news, tfs, ps)
	if err != nil {
		return nil, nil, err
	}
	return inputs, cctx.Assets, err
}
```

That turns the body of `MakeTerraformInputsNoTFDefaults` into:
```go
func MakeTerraformInputsNoTFDefaults(
	ctx context.Context, instance *PulumiResource, config resource.PropertyMap,
	olds, news resource.PropertyMap, tfs shim.SchemaMap, ps map[string]*SchemaInfo,
) (map[string]interface{}, AssetTable, error) {
	return makeTerraformInputsInner(ctx, instance, config,
	olds, news, tfs shim.SchemaMap, ps, false)
}
```

We can do the same thing with `MakeTerraformInputs`.

Reply:
Thanks for the suggestion!